plugins/file-open-save-new/core/src/test/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProviderTest.java

```diff
@@ -0,0 +1,215 @@
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some_Connection" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some-Connection" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection123" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://123someConnection123" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some_Connection/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some-Connection/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection123/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://123someConnection123/" ) ) );
+
+    // TEST now we have past the root domain
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection/someFolderA" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot(
+        createTestInstance( "pvfs://someConnection/someFolderA/directory2" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot(
+        createTestInstance( "pvfs://someConnection/someFolderA/directory2/randomFileC.txt" ) ) );
+  }
+
+  public void testHasBuckets() {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    // TEST non VFSConnectionDetails
+    ConnectionDetails mockConnectionDetails  = mock( ConnectionDetails.class );
+    assertFalse( testInstance.hasBuckets( mockConnectionDetails ) );
+
+    // TEST  does not have buckets
+    VFSConnectionDetails mock_NoBuckets_VFSConnectionDetails  = mock( VFSConnectionDetails.class );
+    when( mock_NoBuckets_VFSConnectionDetails.hasBuckets() ).thenReturn( false );
+    assertFalse( testInstance.hasBuckets( mock_NoBuckets_VFSConnectionDetails ) );
+
+    // TEST has buckets
+    VFSConnectionDetails mock_Buckets_VFSConnectionDetails  = mock( VFSConnectionDetails.class );
+    when( mock_Buckets_VFSConnectionDetails.hasBuckets() ).thenReturn( true );
+    assertTrue( testInstance.hasBuckets( mock_Buckets_VFSConnectionDetails ) );
+  }
+
+  @Test
+  public void testIsSupported() throws Exception {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    assertFalse( testInstance.isSupported( null ) );
+
+    assertFalse( testInstance.isSupported( "" ) );
+
+    assertFalse( testInstance.isSupported( "    " ) );
+
+    assertFalse( testInstance.isSupported( "someGarbage" ) );
+
+    assertFalse( testInstance.isSupported( "/someUser/someUnixFile" ) );
+
+    assertFalse( testInstance.isSupported( "T:\\Users\\RandomSUser\\Documents\\someWindowsFile" ) );
+
+    assertFalse( testInstance.isSupported( "//home/randomUser/randomFile.rpt" ) );
+
+    assertTrue( testInstance.isSupported( "pvfs://someConenction/someFilePath" ) );
```

Comment:
No need to fix.
Typo in `someConenction`

Reply:
https://github.com/pentaho/pentaho-kettle/commit/8d2d79b16445a6dfdaf944f34a702997b3bfe6b5